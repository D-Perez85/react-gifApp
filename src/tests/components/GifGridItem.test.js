import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";
import "@testing-library/jest-dom";

/*1. Mostrar el componente GifGridItem correctamente 
    -shallow
    -wrapper
    -wrapper.toMatchSnapshot(); 
*/

describe("Pruebas en componente GifGridItem", () => {
 
  const title = 'un titulo'; 
  const url = 'una url'; 
  let wrapper = shallow(<GifGridItem title={title} url={url} />);  //WRAPPER GLOBAL

  test("Snapshot de GifGridItem ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el titulo ', () => {
    const p = wrapper.find('p'); 
    expect(p.text().trim()).toBe(title); 
  })
  
  test('La Img deberia de tener una url y un texto alternativo ', () => {
    const img = wrapper.find('img'); 
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })

  test('debe de tener incluido la clase animate__fadeIn', () => {
    const div = wrapper.find('div'); 
    const className = div.prop('className')
    expect(className.includes('animate__fadeIn')).toBe(true)
  })
  
  

});