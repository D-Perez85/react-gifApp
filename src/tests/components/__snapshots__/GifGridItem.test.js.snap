// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem Snapshot de GifGridItem  1`] = `
<div
  className=" card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="una url"
  />
  <p>
    un titulo
  </p>
</div>
`;
